spec/automation/unit/method_validation/dialog_parser_spec.rb

```diff
@@ -55,6 +55,34 @@ def create_tags
       expect(depts).to match_array(@dept_array)
     end
 
+    it "with password option" do
+      dialog_hash = {'password::dialog_option_1_passwordtest' => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}"}
+      parsed_dialog_options_hash = {1 => {:"password::dialog_passwordtest" => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}",
+                                          :"password::passwordtest"        => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}"}}
+
+      @root_stp.options = @root_stp.options.merge(:dialog => dialog_hash)
+      @root_stp.save
+      run_automate_method
+      @root_stp.reload
+
+      pdo = YAML.load(@root_stp.get_option(:parsed_dialog_options))
+      expect(pdo).to eql(parsed_dialog_options_hash)
+    end
+
+    it "with generic password" do
+      dialog_hash = {'password::dialog_passwordtest' => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}"}
+      parsed_dialog_options_hash = {0 => {:"password::dialog_passwordtest" => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}",
+                                          :"password::passwordtest"        => "v2:{i7uvqmb1Dr6WAxCpakNE9w==}"}}
+
+      @root_stp.options = @root_stp.options.merge(:dialog => dialog_hash)
```

Comment:
@tinaafitz 
Can we put line 63 thru 69 in a function and pass in the dialog_hash and parsed_dialog_option_hash into it